Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -5,13 +5,30 @@ namespace Mantid {
 namespace Beamline {
 
 DetectorInfo::DetectorInfo(const size_t numberOfDetectors)
-    : m_isMasked(Kernel::make_cow<std::vector<bool>>(numberOfDetectors)) {}
+    : m_isMonitor(Kernel::make_cow<std::vector<bool>>(numberOfDetectors)),
+      m_isMasked(Kernel::make_cow<std::vector<bool>>(numberOfDetectors)) {}
+
+DetectorInfo::DetectorInfo(const size_t numberOfDetectors,
+                           const std::vector<size_t> &monitorIndices)
+    : DetectorInfo(numberOfDetectors) {
+  for (const auto i : monitorIndices)
+    m_isMonitor.access().at(i) = true;
+}
 
 /// Returns the size of the DetectorInfo, i.e., the number of detectors in the
 /// instrument.
-size_t DetectorInfo::size() const { return m_isMasked->size(); }
+size_t DetectorInfo::size() const {
+  if (!m_isMasked)
```

Comment:
How do you end up storing a null pointer for the masks? I don't see how that is currently allowed.